upgrade/upgrade-scripts/execute

```diff
@@ -353,7 +353,17 @@ zcat "/usr/share/doc/delphix-entire-$platform/packages.list.gz" |
 KVERS=$(apt-mark showauto 'delphix-kernel-.*' | sed 's/^delphix-kernel-//') ||
 	die "failed to determine list of kernel versions to remove"
 for KVER in $KVERS; do
-	apt_get purge -y "delphix-kernel-$KVER" "linux-image-$KVER" ||
+	apt_get purge -y \
+		"delphix-kernel-$KVER" \
+		"linux-image-$KVER" \
+		"linux-image-$KVER-dbgsym" \
+		"linux-modules-extra-$KVER" \
+		"linux-headers-$KVER" \
+		"linux-tools-$KVER" \
+		"zfs-modules-$KVER" \
+		"zfs-modules-$KVER-dbg" \
+		"zfs-headers-$KVER" \
+		"connstat-module-$KVER" ||
```

Comment:
Is there any way to auto-generate this list? My concern is that if a new kernel package gets introduced, it's likely to not be included here until we realize that old versions are piling up on customer systems.

Reply:
Possibly.. it might be reasonable to parse this output:
```
delphix@dcol2:~/ps$ apt-cache depends delphix-kernel-5.4.0-70-dx2021040621-9e69cb9ee-generic
delphix-kernel-5.4.0-70-dx2021040621-9e69cb9ee-generic
  Depends: linux-image-5.4.0-70-dx2021040621-9e69cb9ee-generic
  Depends: linux-image-5.4.0-70-dx2021040621-9e69cb9ee-generic-dbgsym
  Depends: linux-modules-extra-5.4.0-70-dx2021040621-9e69cb9ee-generic
  Depends: linux-headers-5.4.0-70-dx2021040621-9e69cb9ee-generic
  Depends: linux-tools-5.4.0-70-dx2021040621-9e69cb9ee-generic
  Depends: zfs-modules-5.4.0-70-dx2021040621-9e69cb9ee-generic
  Depends: zfs-modules-5.4.0-70-dx2021040621-9e69cb9ee-generic-dbg
  Depends: zfs-headers-5.4.0-70-dx2021040621-9e69cb9ee-generic
  Depends: connstat-module-5.4.0-70-dx2021040621-9e69cb9ee-generic

```
e.g. and remove all packages that `delphix-kernel` depends on...

Reply:
maybe something like this:
```
$ dpkg-query -Wf '${Depends}' delphix-kernel-5.4.0-70-dx2021040621-9e69cb9ee-generic | sed 's/, / /g'
linux-image-5.4.0-70-dx2021040621-9e69cb9ee-generic linux-image-5.4.0-70-dx2021040621-9e69cb9ee-generic-dbgsym linux-modules-extra-5.4.0-70-dx2021040621-9e69cb9ee-generic linux-headers-5.4.0-70-dx2021040621-9e69cb9ee-generic linux-tools-5.4.0-70-dx2021040621-9e69cb9ee-generic zfs-modules-5.4.0-70-dx2021040621-9e69cb9ee-generic zfs-modules-5.4.0-70-dx2021040621-9e69cb9ee-generic-dbg zfs-headers-5.4.0-70-dx2021040621-9e69cb9ee-generic connstat-module-5.4.0-70-dx2021040621-9e69cb9ee-generic
```
?

Reply:
Yeah, that appears to do the trick.